### PR TITLE
docs - using dblink as a non-superuser

### PR DESCRIPTION
**Pull request by mkiyama**
-update installing dblink on 6.0 - uses create extension
-clarify some dblink_connect() information
-add using dblink_connect_u() as a non-superuser

Will be backported to 5X_STABLE

HTML format on a GPDB 6 doc test site.
http://docs-msk-gpdb6-dev.cfapps.io/600/utility_guide/dblink.html

### CODE REVIEW

File: gpdb-doc/dita/utility_guide/dblink.xml [hunk 1]
```diff
@@ -30,82 +38,122 @@ INSERT 0 2</codeblock></p>
...
+testdb=# <b>GRANT EXECUTE ON FUNCTION dblink_connect_u(text, text) TO test_user;</b></codeblock></li>
+          <li>Now <codeph>test_user</codeph> can create a connection to a another local database
+            without a password. For example, <codeph>test_user</codeph> can log into the
+              <codeph>testdb</codeph> database and execute this command to create a connection named
+              <codeph>testconn</codeph> to the local <codeph>postgres</codeph> database.<codeblock>testdb=> <b>SELECT dblink_connect_u('testconn', 'dbname=postgres user=test_user');</b></codeblock>
+            <note>If a <codeph>user</codeph> is not specified, <codeph>dblink_connect_u()</codeph>
+              uses the value of the <codeph>PGUSER</codeph> environment variable when Greenplum
+              Database was started. </note></li>
+          <li><codeph>test_user</codeph> can use the <codeph>dblink()</codeph> function to execute a
+            query using a <codeph>dblink</codeph> connection. For example, this command that uses
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/dblink.xml]: that uses -> uses

File: gpdb-doc/dita/utility_guide/dblink.xml [hunk 2]
```diff
@@ -30,82 +39,122 @@ INSERT 0 2</codeblock></p>
...
-            make a connection to a remote database system, simply include host and port information
-            in the connection string. For example, to create an implicit <codeph>dblink</codeph>
-            connection to a remote
-            system:<codeblock>gpadmin=# <b>SELECT dblink_connect('host=remotehost port=5432 dbname=postgres');</b></codeblock></p><note>You
-            must specify both a hostname and a password in the connection string to connect as a
-            non-superuser.</note></li>
-        <li>Use the basic <codeph>dblink()</codeph> function to query a database using a configured
+(1 row)</codeblock>
+          <note>If a <codeph>user</codeph> is not specified, <codeph>dblink_connect()</codeph> uses
+            the value of the <codeph>PGUSER</codeph> environment variable when Greenplum Database
```
Comment (lisakowen) [inline gpdb-doc/dita/utility_guide/dblink.xml]: this confuses me.   does the PGUSER environment variable need to be set before starting greenplum?

File: gpdb-doc/dita/utility_guide/dblink.xml [hunk 3]
```diff
@@ -30,82 +39,122 @@ INSERT 0 2</codeblock></p>
...
+            server environment, such as a <codeph>~/.pgpass</codeph> file belonging to the server's
+            user.</p></note>
+        <ol id="ol_dpt_ll3_5fb">
+          <li>As a superuser, grant the <codeph>EXECUTE</codeph> privilege on the
+              <codeph>dblink_connect_u()</codeph> functions in the user database. This example
+            grants the privilege to the non-superuser <codeph>test_user</codeph> on the functions
+            with the signatures for creating an implicit or a named <codeph>dblink</codeph>
+            connection.<codeblock>testdb=# <b>GRANT EXECUTE ON FUNCTION dblink_connect_u(text) TO test_user;</b>
+testdb=# <b>GRANT EXECUTE ON FUNCTION dblink_connect_u(text, text) TO test_user;</b></codeblock></li>
+          <li>Now <codeph>test_user</codeph> can create a connection to a another local database
```
Comment (lisakowen) [inline gpdb-doc/dita/utility_guide/dblink.xml]: "a another" -> another

File: gpdb-doc/dita/utility_guide/dblink.xml [hunk 4]
```diff
@@ -30,82 +39,122 @@ INSERT 0 2</codeblock></p>
...
+            without a password. For example, <codeph>test_user</codeph> can log into the
+              <codeph>testdb</codeph> database and execute this command to create a connection named
+              <codeph>testconn</codeph> to the local <codeph>postgres</codeph> database.<codeblock>testdb=> <b>SELECT dblink_connect_u('testconn', 'dbname=postgres user=test_user');</b></codeblock>
+            <note>If a <codeph>user</codeph> is not specified, <codeph>dblink_connect_u()</codeph>
```
Comment (lisakowen) [inline gpdb-doc/dita/utility_guide/dblink.xml]: same comment as above re: PGUSER

File: gpdb-doc/dita/utility_guide/dblink.xml [hunk 5]
```diff
@@ -30,82 +39,122 @@ INSERT 0 2</codeblock></p>
...
+              uses the value of the <codeph>PGUSER</codeph> environment variable when Greenplum
+              Database was started. </note></li>
+          <li><codeph>test_user</codeph> can use the <codeph>dblink()</codeph> function to execute a
+            query using a <codeph>dblink</codeph> connection. For example, this command uses the
+              <codeph>dblink</codeph> connection named <codeph>testconn</codeph> created in the
+            previous
+            step.<codeblock>testdb=> <b>SELECT * FROM dblink('testconn', 'SELECT * FROM testdblink') AS dbltab(id int, product text);</b></codeblock></li>
```
Comment (lisakowen) [inline gpdb-doc/dita/utility_guide/dblink.xml]: do we want to mention that test_user must have been granted explicit permission to access the table named testdblink?  (not related to dblink, but the superuser created the table in step 1 of the previous section.)

**Pull request merged**